internal/gatewayapi/resource/load.go

```diff
@@ -384,22 +385,21 @@ func loadKubernetesYAMLToResources(input []byte, addMissingResources bool) (*Res
 			providedNamespaceMap.Insert(config.DefaultNamespace)
 		}
 	}
-
-	if addMissingResources {
-		// Sort the required namespace in order to keep the consistency for test.
-		sortedRequiredNamespace := requiredNamespaceMap.UnsortedList()
-		sort.Strings(sortedRequiredNamespace)
-		for _, ns := range sortedRequiredNamespace {
-			if !providedNamespaceMap.Has(ns) {
-				namespace := &corev1.Namespace{
-					ObjectMeta: metav1.ObjectMeta{
-						Name: ns,
-					},
-				}
-				resources.Namespaces = append(resources.Namespaces, namespace)
+	// Sort the required namespace in order to keep the consistency for test.
```

Comment:
Nit: a short comment here would help clarify why namespaces should always be added, even when `addMissingResources` is not set to true.

Reply:
Added